src/media/js/apps_buttons.js

```diff
@@ -47,14 +47,27 @@ define('apps_buttons',
 
         // If it's a paid app, ask the user to sign in first.
         if (product.receipt_required && !user.logged_in()) {
+            var payWindow;
+
+            if (!capabilities.navPay) {
+                console.log('Creating payment window ahead of time via user click');
+                payWindow = payments.utils.openWindow({url: ''});
```

Comment:
The downside of this is that the context is then wrong for the defaults usage (a bug in itself). I'll leave this as is and fix that up next week along with the default url.